hasher-matcher-actioner/hmalib/lambdas/pdq/pdq_matcher.py

```diff
@@ -44,12 +59,19 @@ def get_index(bucket_name, key):
 
 def lambda_handler(event, context):
     """
-    Listens to SQS events fired when new hash is generated. Loads the index stored in
-    an S3 bucket and looks for a match
+    Listens to SQS events fired when new hash is generated. Loads the index
+    stored in an S3 bucket and looks for a match.
 
     As per the default configuration
     - the index data bucket is INDEXES_BUCKET_NAME
     - the key name must be PDQ_INDEX_KEY
+
+    When matched, publishes a notification to an SNS endpoint. Note this is in
+    contrast with hasher and indexer. They publish to SQS directly. Publishing
+    to SQS implies there can be only one consumer.
+
+    Because, here, in the matcher, we publish to SNS, we can plug multiple
+    queues behind it and profit!
```

Comment:
An SNS topic can have any number of subscriptions. Right now, the only subscription is an SQS queue, but partners could (using the console) configure other subscriptions (email, firehose, http).

You wouldn't route to a specific queue, more like double process. eg. Match happened, one queue plugs in to actioning lambdas, the other queue plugs in to record-keeping / enrichment lambdas. or drops it to a review queue.

Reply:
> @schatten - I still think we potentially need this topic, because the SEEN writeback might be determined at the matcher level (though it could also work from the decision making level, which just then becomes an implicit rule of (level for bank with match) -> Action for seen.

In an offline discussion, @jeberl @eriksendc and I came to the same conclusion. The SEEN writeback would be the first case of branching out from the match SNS notification. As Jesse has hinted, 

```
--> matcher λ --> SNS Topic --> actioner queue --> actioner λ
                          \--> logging queue --> seen writeback λ
```